flink-autoscaler-standalone/src/test/java/org/apache/flink/autoscaler/standalone/StandaloneAutoscalerExecutorTest.java

```diff
@@ -0,0 +1,122 @@
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.fail;
+
+/** Test for {@link StandaloneAutoscalerExecutor}. */
+class StandaloneAutoscalerExecutorTest {
+
+    @ParameterizedTest
+    @ValueSource(booleans = {true, false})
+    void testScaling(boolean throwExceptionWhileScale) {
+        JobAutoScalerContext<JobID> jobContext1 = createJobAutoScalerContext();
+        JobAutoScalerContext<JobID> jobContext2 = createJobAutoScalerContext();
+        var jobList = List.of(jobContext1, jobContext2);
+        Set<JobID> exceptionKeys =
+                throwExceptionWhileScale
+                        ? Set.of(jobContext1.getJobKey(), jobContext2.getJobKey())
+                        : Set.of();
+
+        var actualScaleContexts = new ArrayList<JobAutoScalerContext<JobID>>();
+
+        var eventCollector = new TestingEventCollector<JobID, JobAutoScalerContext<JobID>>();
+        var autoscalerExecutor =
+                new StandaloneAutoscalerExecutor<>(
+                        Duration.ofSeconds(2),
+                        () -> jobList,
+                        eventCollector,
+                        createJobAutoScaler(actualScaleContexts, exceptionKeys));
+
+        autoscalerExecutor.scaling();
+        assertThat(actualScaleContexts).isEqualTo(jobList);
+        assertThat(eventCollector.events)
+                .hasSameSizeAs(exceptionKeys)
+                .allMatch(
+                        event ->
+                                event.getReason()
+                                        .equals(StandaloneAutoscalerExecutor.AUTOSCALER_ERROR));
+    }
+
+    @Test
+    void testFetchException() {
+        var eventCollector = new TestingEventCollector<JobID, JobAutoScalerContext<JobID>>();
+        var autoscalerExecutor =
+                new StandaloneAutoscalerExecutor<>(
+                        Duration.ofSeconds(2),
+                        () -> {
+                            throw new RuntimeException("Excepted exception.");
+                        },
+                        eventCollector,
+                        new JobAutoScaler<>() {
+                            @Override
+                            public void scale(JobAutoScalerContext<JobID> context) {
+                                fail("Should be called.");
+                            }
+
+                            @Override
+                            public void cleanup(JobID jobID) {
+                                fail("Should be called.");
+                            }
+                        });
+
+        // scaling shouldn't throw exception even if fetch fails
+        autoscalerExecutor.scaling();
```

Comment:
Technically, I don't know why need this. Because the test will fail if `autoscalerExecutor.scaling();` throw any exception.

From a code design perspective, explicitly declaring `doesNotThrow` may be more clear for code maintenance. So it sounds good to me. I updated.